.gitlab-ci.yml

```diff
@@ -0,0 +1,36 @@
+stages:
+  - benchmark
+
+variables:
+  PYTHONUNBUFFERED: "true"
+  JVMCI_VERSION_CHECK: ignore
+  ECLIPSE_EXE: /home/gitlab-runner/.local/eclipse/eclipse
+  JAVA_HOME: /usr/lib/jvm/java-8-openjdk-amd64
+
+before_script:
+  - git submodule update --init
+
+# I'll disable it just for now, I haven't fixed checkstyle in my IDE yet
```

Comment:
Did you try: https://plugins.jetbrains.com/plugin/1065-checkstyle-idea

If it crashes, you may need to update the checkstyle config. In SOMns, it should already be updated: https://github.com/smarr/SOMns/commit/684aa34b6cc397449b7b0e8fc1119914908fef9c

Reply:
I should re-enable it actually, I installed this plugin a while back and it works. Some bits of the checkstyle configuration files from SOMns do not work with it (but on not-so-relevant bits such as formatting of tests/copyright headers...) but overall it does the job